### ISSUE TITLE
break info/warning/error/debug messages into log levels 

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/stan#512]
(Originally suggested by John Hall on stan-users;  my words in the issue.)

Given the number of users we've confused with the rejection warnings, it would be nice to have levels of reporting like a typical logger.  I'm most familiar with log4j, which has messages of increasing verbosity, which are roughly  error, warning, info, debug (see <a href="http://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/lf5/LogLevel.html">LogLevel javadoc</a>).  

This would allow us to turn off the warnings most of the time and only turn them on if a model's having problems.

**Comment by syclik** [comment]
I'll start a wiki on what I think the design should be. I think @sakrejda and I already started one, so maybe I'll dig that up.

This depends on #1751, but once we're past that, we can start moving on this.
